core/src/main/java/com/linecorp/armeria/common/DefaultFlagsProvider.java

```diff
@@ -0,0 +1,326 @@
+        return 1000000L;
+    }
+
+    @Override
+    public Integer defaultHttp2InitialConnectionWindowSize() {
+        return 1024 * 1024; // 1MiB
+    }
+
+    @Override
+    public Integer defaultHttp2InitialStreamWindowSize() {
+        return 1024 * 1024; // 1MiB
+    }
+
+    @Override
+    public Integer defaultHttp2MaxFrameSize() {
+        return 16384; // From HTTP/2 specification
+    }
+
+    @Override
+    public Long defaultHttp2MaxStreamsPerConnection() {
+        return Long.valueOf(Integer.MAX_VALUE);
+    }
+
+    @Override
+    public Long defaultHttp2MaxHeaderListSize() {
+        return 8192L; // from Netty default maxHeaderSize
+    }
+
+    @Override
+    public String defaultBackoffSpec() {
+        return "exponential=200:10000,jitter=0.2";
+    }
+
+    @Override
+    public Integer defaultMaxTotalAttempts() {
+        return 10;
+    }
+
+    @Override
+    public String routeCacheSpec() {
+        return "maximumSize=4096";
+    }
+
+    @Override
+    public String routeDecoratorCacheSpec() {
+        return "maximumSize=4096";
+    }
+
+    @Override
+    public String parsedPathCacheSpec() {
+        return "maximumSize=4096";
+    }
+
+    @Override
+    public String headerValueCacheSpec() {
+        return "maximumSize=4096";
+    }
+
+    @Override
+    public List<String> cachedHeaders() {
+        return Splitter.on(',').trimResults()
+                       .omitEmptyStrings()
+                       .splitToList(":authority,:scheme,:method,accept-encoding,content-type");
+    }
+
+    @Override
+    public String fileServiceCacheSpec() {
+        return "maximumSize=1024";
+    }
+
+    @Override
+    public String dnsCacheSpec() {
+        return "maximumSize=4096";
+    }
+
+    @Override
+    public Predicate<InetAddress> preferredIpV4Addresses() {
+        return null;
+    }
+
+    @Override
+    public Boolean useJdkDnsResolver() {
+        return false;
+    }
+
+    @Override
+    public Boolean reportBlockedEventLoop() {
+        return true;
+    }
+
+    @Override
+    public Boolean validateHeaders() {
+        return true;
+    }
+
+    @Override
+    public Boolean tlsAllowUnsafeCiphers() {
+        return false;
+    }
+
+    @Override
+    public Set<TransientServiceOption> transientServiceOptions() {
+        return ImmutableSet.of();
+    }
+
+    @Override
+    public Boolean useDefaultSocketOptions() {
+        return true;
+    }
+
+    @Override
+    public Boolean useLegacyRouteDecoratorOrdering() {
+        return false;
+    }
+
+    @Override
+    public Boolean allowDoubleDotsInQueryString() {
+        return false;
+    }
+
+    @Override
+    public Path defaultMultipartUploadsLocation() {
+        return Paths.get(System.getProperty("java.io.tmpdir") +
+                         File.separatorChar + "armeria" +
+                         File.separatorChar + "multipart-uploads");
+    }
+
+    private DefaultFlagsProvider() {}
```

Comment:
Should we move the constructor up to member methods?
https://armeria.dev/community/developer-guide#organize